src/lib/helpers/helpers.ts

```diff
@@ -0,0 +1,19 @@
+function distinct(array: any[], comparator?: (value1: any, value2: any) => number): any[] {
+	if (comparator) {
+		return distinctWithComparator(array, comparator);
+	} else {
+		return distinctWithoutComparator(array);
+	}
+}
+
+function distinctWithComparator(array: any[], comparator: (value1: any, value2: any) => number): any[] {
+	return array.sort(comparator).filter((value, index, array) => index === 0 || comparator(value, array[index - 1]) !== 0);
```

Comment:
can it be removed here as well so the behaviour is consistent?

Reply:
not really unless you want this to not work. Also the sort I removed shouldn't be needed with the new distinct() implementation

Reply:
> unless you want this to not work

rofl

Reply:
Might be misunderstanding, but does one function return an unsorted list and the other not now? 

I think either have both return sorted for consistency or look at what needs to be done to make both preserve the original order. https://stackoverflow.com/a/15868720 seems to have some suggestions (one of the comments suggests that it has the ability to manipulate objects?). That or lo-dash or underscore.js might have solutions? 

I'm not too stressed about it, but whichever way we go I think it should be consistent and clear

Reply:
I don't think anyone should expect a distinct function to return the items in any particular order (sorted or unsorted). It does what it says on the tin, so personally I'm happy with it.

Reply:
I'm just saying whether you hit the comparator version or the plain version, it should probably return consistently sorted, or preserved order array.

Reply:
As long as there's no claim that it's a stable distinct function, I don't think it matters whether or not it maintains any particular order.

The contract is just that items will be distinct according to the comparator, nothing more.